internal/controllers/image/upload.go

```diff
@@ -83,11 +83,6 @@ func (r *orcImageReconciler) uploadImageContent(ctx context.Context, orcImage *o
 	}
 
 	download := content.Download
-	if download == nil {
-		// Should have been caught by validation
-		return orcerrors.Terminal(orcv1alpha1.ConditionReasonInvalidConfiguration, "image source type URL has no url entry")
-	}
```

Comment:
It's still a pointer, so we still need this.

Reply:
Done